pkg/local-resource-manager/controller_manager.go

```diff
@@ -38,6 +40,108 @@ import (
 // resources calculation.
 
 // Start starts the controller.
+
+func canMergeFlavours(flavour1, flavour2 *nodecorev1alpha1.Flavour) bool {
+	if flavour1.Spec.ProviderID != flavour2.Spec.ProviderID {
+		return false
+	}
+
+	if flavour1.TypeMeta != flavour2.TypeMeta {
+		return false
+	}
+
+	if flavour1.Namespace != flavour2.Namespace {
+		return false
+	}
+
+	if flavour1.Spec.OptionalFields.Availability != flavour2.Spec.OptionalFields.Availability {
+		return false
+	}
+
+	if flavour1.Spec.Owner != flavour2.Spec.Owner {
+		return false
+	}
+
+	if flavour1.Spec.Price != flavour2.Spec.Price {
+		return false
+	}
+
+	if flavour1.Spec.Type != flavour2.Spec.Type {
+		return false
+	}
+
+	if !reflect.DeepEqual(flavour1.Spec.Policy, flavour2.Spec.Policy) {
+		return false
+	}
+
+	if flavour1.Spec.Characteristics.Architecture != flavour2.Spec.Characteristics.Architecture {
+		return false
+	}
+
+	if flavour1.Spec.Characteristics.Gpu != flavour2.Spec.Characteristics.Gpu {
+		return false
+	}
+
+	if flavour1.Spec.Characteristics.Pods != flavour2.Spec.Characteristics.Pods {
+		return false
+	}
+
+	if flavour1.Spec.Characteristics.Cpu.Value() != flavour2.Spec.Characteristics.Cpu.Value() {
+		return false
+	}
+
+	if flavour1.Spec.Characteristics.Memory.Value()/
+		flavour1.Spec.Policy.Partitionable.MemoryStep.Value() !=
+		flavour2.Spec.Characteristics.Memory.Value()/
+			flavour2.Spec.Policy.Partitionable.MemoryStep.Value() {
+		return false
+	}
+
+	if flavour1.Spec.Characteristics.EphemeralStorage.Value() != flavour2.Spec.Characteristics.EphemeralStorage.Value() {
+		return false
+	}
+
+	if flavour1.Spec.Characteristics.PersistentStorage.Value() != flavour2.Spec.Characteristics.PersistentStorage.Value() {
+		return false
+	}
+
+	return true
+}
+
+func mergeFlavours(flavours []nodecorev1alpha1.Flavour) []nodecorev1alpha1.Flavour {
+	mergedFlavours := []nodecorev1alpha1.Flavour{}
+
+	for len(flavours) > 0 {
+		mergedFlavour := flavours[0]
+		mergedFlavour.Spec.QuantityAvailable = 1
+
```

Comment:
If we want to support Aggregation, what do we mean by that?

I thought we needed to combine different flavors into a big one.

E.g. Aggregated Flavor CPU = Flavor 1 CPU + Flavor 2 CPU; Aggregated Flavor RAM = Flavor 1 RAM + FLAVOR 2 RAM, etc.

Am I missing something?

Reply:
The aggregation of heterogeneus resources could be another kind of aggregation but we need to specify and design it.

Reply:
What do you mean by "heterogeneous resources"? 
Resources of the same "type" but with different characteristics? Or resources of different type?

Reply:
Resources of the same type but different characteristics.